### PR TITLE
Creating an ARM template for Cosmos DB SQL data plane RBAC

### PR DESCRIPTION
**Pull request by nichatur**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Adding a template to create a SQL Cosmos account, a natively maintained Role Definition, and a natively maintained Role Assignment for an AAD identity.

### CODE REVIEW

File: 101-cosmosdb-sql-rbac/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for all resources."
+      }
+    },
+    "accountName": {
```
Comment (bmoore-msft) [inline 101-cosmosdb-sql-rbac/azuredeploy.json:12]: add a defaultValue with uniqueString

File: 101-cosmosdb-sql-rbac/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      "type": "string",
+      "metadata": {
+        "description": "Cosmos DB account name, max length 44 characters"
+      }
+    },
+    "roleDefinitionId": {
+      "type": "string",
+      "defaultValue": "[newGuid()]",
```
Comment (bmoore-msft) [inline 101-cosmosdb-sql-rbac/azuredeploy.json:20]: this must be idempotent

File: 101-cosmosdb-sql-rbac/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        "Microsoft.DocumentDB/databaseAccounts/readMetadata",
+        "Microsoft.DocumentDB/databaseAccounts/sqlDatabases/containers/items/*"
+      ],
+      "metadata": {
+        "description": "Data actions permitted by the Role Definition"
+      }
+    },
+    "roleAssignmentId": {
+      "type": "string",
+      "defaultValue": "[newGuid()]",
```
Comment (bmoore-msft) [inline 101-cosmosdb-sql-rbac/azuredeploy.json:44]: this must be idempotent

File: 101-cosmosdb-sql-rbac/azuredeploy.parameters.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "accountName": {
+            "value": "GEN-UNIQUE"
+        },
+        "principalId": {
+            "value": "GEN-AZUREAD-OBJECTID"
+        },
+        "location": {
```
Comment (bmoore-msft) [inline 101-cosmosdb-sql-rbac/azuredeploy.parameters.json:11]: remove this and use the defaultValue in the template

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

**Comment by nichatur** [comment]
Hi @bmoore-msft , Thanks for much for your latest commit! 
I do have one worry however, if a customer uses this ARM template for regular deployment and decides to update the "dataActions", will this not lead to the creation of a new Role Definition instead of simply updating the existing one?

Especially because we no longer provide the ability to the customer to specify this ID via a parameter to ensure it stays consistent across deployments.

**Comment by bmoore-msft** [comment]
If a roleDefinition (or roleAssignment) is created with the same properties and a different name/guid - the PUT will fail, so using a random guild will eventually block deployment.

**Pull request merged**